xdsl/dialects/scf.py

```diff
@@ -50,6 +58,75 @@ def get(cond: SSAValue | Operation,
             operands=[cond, [output for output in output_ops]])
 
 
+@irdl_op_definition
+class For(Operation):
+    name: str = "scf.for"
+
+    lb: Annotated[Operand, IndexType]
+    ub: Annotated[Operand, IndexType]
+    step: Annotated[Operand, IndexType]
+
+    iter_args: Annotated[VarOperand, AnyAttr()]
+
+    res: Annotated[VarOpResult, AnyAttr()]
+
+    body: Region
+
+    def verify_(self):
+        if (len(self.iter_args) + 1) != len(self.body.blocks[0].args):
+            raise VerifyException(
+                f"Wrong number of block arguments, expected {len(self.iter_args)+1}, got "
+                f"{len(self.body.blocks[0].args)}. The body must have the induction "
+                f"variable and loop-carried variables as arguments.")
+        if len(self.operands) < 3:
+            raise VerifyException(
+                "Expected at least 3 arguments. A scf.for must have a lower- and "
+                "upper-bound, and a step.")
+        for idx, arg in enumerate(self.iter_args):
+            if self.body.blocks[0].args[idx + 1].typ != arg.typ:
+                raise VerifyException(
+                    f"Block arguments with wrong type, expected {arg.typ}, "
+                    f"got {self.body.blocks[0].args[idx].typ}. Arguments after the "
+                    f"induction variable must match the carried variables.")
+        if len(self.iter_args) > 0:
+            if len(self.body.blocks[0].ops) == 0 or not isinstance(
```

Comment:
Can you put parentheses around the entire condition? Just to see if it makes it prettier (so the isinstance call is not away)